Factory/FilterFactory.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Epam.NetMentoring.Factory
+{
+    class FilterFactory : IFilterFactory
+    {
+        public virtual IFilter CreateFilter(string filtername)
+        {
+            if (filtername == "BOFA")
```

Comment:
We can consider 'switch' instead of multiple if-else, looks more clear and less typing?

Reply:
ok